### PR TITLE
Reflect the service nullability behavior of Error and endTime

### PR DESCRIPTION
**Pull request by christothes**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by heaths — COMMENTED** [review]
LGTM, but I'd rather the service team sign-off on the changes.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json [hunk 1]
```diff
@@ -67,7 +67,8 @@
           "$ref": "#/definitions/Error"
         }
       },
-      "description": "The key vault server error."
+      "description": "The key vault server error.",
+      "x-nullable": true
```
Comment (jhendrixMSFT) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: @johanste is this correct?  I thought `x-nullable` would be applied where this type is returned.

Comment (johanste) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: Does the service actually return `null` here? If so, why? The property is optional. And we really don't want to have semantically meaningful `null` values for content-type application/json payloads.

Comment (christothes) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: In my testing the service returns null for properties of the `Error` model, including `innererror`, which is of type `Error`.

https://github.com/Azure/azure-rest-api-specs/blob/625a4165f8a2ee84549f61e46620e1442a509ab5/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json#L64-L68

**Review by vasanthrajams — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json (hunk 1 above)
Comment (jhendrixMSFT) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: If the intention is to indicate that fields of the `Error` type can be null then you want to apply `x-nullable` to the applicable fields (like you did in backuprestore.json).

Comment (christothes) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: The error value itself is returned as null. I think the service should clarify which of the types and properties are  expected to be nullable before finalizing on the correct swagger change. 

Currently, marking just the Error properties results in a null ref exception in generated code, but marking the whole Error type does not.

Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: Did you check with @pakrym - or even just try it with local swaggers - if setting `x-nullable: true` on the model itself even works as intended?

**Review by heaths — APPROVED** [review]
Approving to keep unblocked since I'm going OOF soon, but please make sure that marking a model as nullable even works as intended. It's ideal if it does since any operation can return it, so hopefully you don't have to mark all refs as nullable.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json (hunk 1 above)
Comment (christothes) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: Yes to both :)

Comment (vasanthrajams) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: Managed HSM usage of Error is that when there is no error in a correct case we are returning null to indicate the same.

Comment (johanste) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: The service should not return a `null` value on the wire. You should omit the key/value pair altogether. Is there a reason that the service is sending null on the wire?

**Review by jhendrixMSFT — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json (hunk 1 above)
Comment (christothes) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/common.json:71]: In speaking with @vasanthrajams previously, this is consistent with existing Key Vault api behavior, so changing it now could be breaking.

**Pull request merged**